source/opt/aggressive_dead_code_elim_pass.h

```diff
@@ -161,13 +160,13 @@ class AggressiveDCEPass : public MemPass {
   // immediate controlling structured if or loop.  A loop header block points
   // to its own branch instruction.  An if-selection block points to the branch
   // of an enclosing construct's header, if one exists.
-  std::unordered_map<BasicBlock*, Instruction*> block2headerBranch_;
+  CAUnorderedMap<BasicBlock*, Instruction*> block2headerBranch_;
```

Comment:
Do we have to do every single one? Is it every map that is an issue or can we limit this? My fear is we're going to end up with a mix og CAUnorderedMap and std::unordered_map throughout. If we know of the critical sections and just allocate those it will be easier to document what's going on (and make the code change a lot smaller)

Reply:
I think it's safer this way. We'll keep changing the code. Previously time-consuming sections can be less time-consuming, and vice versa. Unless we want to do periodical to adjust, which I think is too burdensome.

Converting `std::*` into `CA*` ones does not hurt, even if it is not very time-consuming.

Using mixed `std::*` ones and `CA*` ones won't be a problem I think. They are different types and the compiler can guarantee that new/delete/copy/etc. is properly handled.

The downside is that we now can use more memory for running a recipe. But it's just a trade-off.

Reply:
I guess my question is, can we get the performance benefit while making this less invasive? True, it might regress in the future, but it seems like the kind of thing we should be testing periodically (or getting some performance tests on the bots would be even better).